src/python/pants/backend/go/tailor.py

```diff
@@ -114,12 +114,8 @@ async def find_putative_go_external_module_targets(
             for search_path in request.search_paths.dirs
         ]
     )
-    candidate_targets = await Get(UnexpandedTargets, AddressSpecs(addresses))
-    go_module_targets = [
-        tgt
-        for tgt in candidate_targets
-        if tgt.has_field(GoModuleSources) and not tgt.address.is_file_target
-    ]
+    candidate_targets = await Get(Targets, AddressSpecs(addresses))
```

Comment:
Even with `indivisable=True`, it still is probably useful I think to use `UnexpandedTargets` because it avoids generating subtargets for all non-Go targets that are covered by these specs.